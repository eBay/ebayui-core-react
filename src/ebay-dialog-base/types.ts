import { KeyboardEvent, SyntheticEvent, MouseEvent } from 'react'

export type DialogCloseEvent =
  | SyntheticEvent<HTMLElement>
  | KeyboardEvent<HTMLElement>
  | MouseEvent<Element>

export type DialogCloseEventHandler = (event: DialogCloseEvent) => void
