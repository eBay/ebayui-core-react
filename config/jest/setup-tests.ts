import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
