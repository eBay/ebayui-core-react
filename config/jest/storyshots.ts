import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
