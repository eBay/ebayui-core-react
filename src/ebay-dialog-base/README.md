# DialogBase

## DialogBase Usage

```react
<DialogBase
    open
    a11yClosetext = "Close DialogBase"
    onCloseBtnClick={ handleCloseBtnClick }>

        <h1>Hello World</h1>

</DialogBase>
```

## Attributes

| Name             | Type                      | Stateful | Required | Description                                                                                  |
|------------------|---------------------------|----------|----------|----------------------------------------------------------------------------------------------|
| `baseEl`         | string                    | No       | No       | ---  div' , 'span', 'aside'                                                                  |
| `open`           | Boolean                   | Yes      | No       | Whether drawer is open.                                                                      |
| `classPrefix`    | drawer or toast or dialog | No       | No       | "" (default) / modifies the base prefix for all  classes in the component                    |
| `windowClass`    | String                    | No       | No       | "" (default) / modifies the base prefix for all Skin classes in the component                |
| `header`         | String                    | No       | No       | element placeholder                                                                          |
| `footer`         | String                    | No       | No       | element placeholder                                                                          |
| `isModal`        | Boolean                   | No       | No       | Whether drawer is model.                                                                     |
| `top`            | ---                       | ---      | ---      | ---                                                                                          |
| `buttonPosition` | string                    | ---      | ---      | top or right or bottom or left                                                               |
| `a11yCloseText`  | String                    | No       | No*      | A11y text for close button and mask. Required only when close button exists.                 |
| `animated`       | Boolean                   | Yes      | No       | Renders the dialog with an animation. Note that the dialog will always be present in the DOM |

## Events

Event | Data | Description
`onCloseBtnClick` | --- | --- | --- | ---
`OnBackgroundClick` | --- | --- | --- | ---
