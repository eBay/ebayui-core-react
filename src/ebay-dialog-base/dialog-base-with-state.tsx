import React, { Children, ReactElement } from 'react'
import { RemoveScroll } from 'react-remove-scroll'
import { DialogBase, DialogBaseProps } from './components/dialogBase'
import { EbayDialogCloseButton, EbayDialogFooter, EbayDialogHeader } from './index'

export const DialogBaseWithState = ({
    isModal,
    open,
    children,
    animated = true,
    ...rest
}: DialogBaseProps<HTMLElement>): ReactElement => {
    const shouldRenderModal = isModal !== false

    const childrenArray = Children.toArray(children) as ReactElement[]
    const header = childrenArray.find((child: ReactElement) => child.type === EbayDialogHeader)
    const footer = childrenArray.find((child: ReactElement) => child.type === EbayDialogFooter)
    const closeButton = childrenArray.find((child: ReactElement) => child.type === EbayDialogCloseButton)
    const content = childrenArray.filter((child: ReactElement) =>
        ![EbayDialogHeader, EbayDialogFooter, EbayDialogCloseButton].some(c => c === child.type))

    const dialogBase = (
        <DialogBase
            {...rest}
            open={open}
            isModal={shouldRenderModal}
            header={header}
            footer={footer}
            closeButton={closeButton}
            animated={animated}
        >
            {content}
        </DialogBase>
    )

    const renderOverLay = () => shouldRenderModal ? (
        <RemoveScroll enabled={open}>
            {dialogBase}
        </RemoveScroll>
    ) : dialogBase

    return animated || open ? renderOverLay() : null
}

export default DialogBaseWithState
