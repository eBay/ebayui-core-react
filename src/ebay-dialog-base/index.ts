export { default as EbayDialogHeader } from './components/dialog-header'
export { default as EbayDialogFooter } from './components/dialog-footer'
export { default as EbayDialogCloseButton } from './components/dialog-close-button'
export { WindowType, DialogBaseProps } from './components/dialogBase'
export { default as DialogBaseWithState } from './dialog-base-with-state'
export { default as EbayDialogActions } from './components/dialog-actions'
