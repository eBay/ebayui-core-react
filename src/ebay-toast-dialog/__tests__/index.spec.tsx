import * as React from 'react';
import requireContext from 'node-require-context'
import { render, fireEvent } from '@testing-library/react'
import { EbayDialogHeader } from '../../ebay-dialog-base'
import { EbayToast } from '../index'
import { initStoryshots } from '../../../config/jest/storyshots'

jest.mock('../../common/random-id', () => ({ randomId: () => 'abc123' }))

describe('<EbayToast>', () => {
    const closeToastHandler = jest.fn()

    beforeEach(() => {
        render(
            <EbayToast
                open
                onClose={closeToastHandler}
            >
                <EbayDialogHeader>Heading</EbayDialogHeader>
                <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                    incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
                    nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </p>
                <p><a href="http://www.ebay.com">www.ebay.com</a></p>
                <input placeholder="Enter a value" />
            </EbayToast>
        )
    })

    it('should have close button', () => {
        expect(document.querySelector('button.toast-dialog__close')).toBeInTheDocument()
    })

    it('should not focus on close button', () => {
        expect(document.querySelector('button.toast-dialog__close')).not.toHaveFocus()
    })

    it('should trigger onClose when close button is clicked', () => {
        fireEvent.click(document.querySelector('button.toast-dialog__close'))
        expect(closeToastHandler).toBeCalled()
    })
})

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/);
        return configure(req, module)
    }

})
