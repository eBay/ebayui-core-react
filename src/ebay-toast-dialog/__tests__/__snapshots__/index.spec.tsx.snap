// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-toast-dialog AlwaysOpened 1`] = `
<DocumentFragment>
  <aside
    aria-labelledby="dialog-title-abc123"
    aria-live="polite"
    arial-modal="true"
    class="toast-dialog toast-dialog--transition toast-dialog--show-init"
    role="dialog"
  >
    <div
      class="toast-dialog__window"
    >
      <div
        class="toast-dialog__header"
      >
        <h2
          id="dialog-title-abc123"
        >
          Toast Dialog Heading
        </h2>
        <button
          aria-label="Close EbayToast"
          class="icon-btn toast-dialog__close icon-btn--transparent"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--close"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
      <div
        class="toast-dialog__main"
      >
        <p>
          Press \`Control-Option-v\` (Mac) to activate CTA button
        </p>
        <p>
          <a
            href="http://www.ebay.com"
          >
            www.ebay.com
          </a>
        </p>
      </div>
      <div
        class="toast-dialog__footer"
      >
        <button
          accesskey="v"
          class="btn btn--primary"
        >
          View Account
        </button>
      </div>
    </div>
  </aside>
  <p>
    For more info, please visit our 
    <a
      href="http://pages.ebay.com/help"
    >
      Help page
    </a>
  </p>
  <p>
    Please check 
    <a
      href="https://www.ebay.de/myb/PurchaseHistory"
    >
      My eBay page
    </a>
     for more details 
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-toast-dialog Default 1`] = `
<DocumentFragment>
  <button
    class="btn btn--secondary"
  >
    Open Toast
  </button>
  <aside
    aria-labelledby="dialog-title-abc123"
    aria-live="polite"
    arial-modal="true"
    class="toast-dialog toast-dialog--transition"
    hidden=""
    role="dialog"
  >
    <div
      class="toast-dialog__window"
    >
      <div
        class="toast-dialog__header"
      >
        <h2
          id="dialog-title-abc123"
        >
          Toast Dialog Heading
        </h2>
        <button
          aria-label="Close EbayToast"
          class="icon-btn toast-dialog__close icon-btn--transparent"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--close"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
      <div
        class="toast-dialog__main"
      >
        <p>
          Press \`Control-Option-c\` (Mac) to activate CTA button
        </p>
        <p>
          <a
            href="http://www.ebay.com"
          >
            www.ebay.com
          </a>
        </p>
      </div>
      <div
        class="toast-dialog__footer"
      >
        <button
          accesskey="c"
          class="btn btn--primary"
        >
          Close Dialog
        </button>
      </div>
    </div>
  </aside>
  <p>
    For more info, please visit our 
    <a
      href="http://pages.ebay.com/help"
    >
      Help page
    </a>
  </p>
  <p>
    Please check 
    <a
      href="https://www.ebay.de/myb/PurchaseHistory"
    >
      My eBay page
    </a>
     for more details 
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-toast-dialog WithAnimation 1`] = `
<DocumentFragment>
  <button
    class="btn btn--secondary"
  >
    Open Toast
  </button>
  <aside
    aria-labelledby="dialog-title-abc123"
    aria-live="polite"
    arial-modal="true"
    class="toast-dialog toast-dialog--transition"
    hidden=""
    role="dialog"
  >
    <div
      class="toast-dialog__window"
    >
      <div
        class="toast-dialog__header"
      >
        <h2
          id="dialog-title-abc123"
        >
          Toast Dialog Heading
        </h2>
        <button
          aria-label="Close EbayToast"
          class="icon-btn toast-dialog__close icon-btn--transparent"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--close"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
      <div
        class="toast-dialog__main"
      >
        <p>
          Toast dialog paragraph text.
        </p>
        <p>
          <a
            href="http://www.ebay.com"
          >
            www.ebay.com
          </a>
        </p>
      </div>
      <div
        class="toast-dialog__footer"
      >
        <button
          class="btn btn--primary"
        >
          View Account
        </button>
      </div>
    </div>
  </aside>
  <p>
    For more info, please visit our 
    <a
      href="http://pages.ebay.com/help"
    >
      Help page
    </a>
  </p>
  <p>
    Please check 
    <a
      href="https://www.ebay.de/myb/PurchaseHistory"
    >
      My eBay page
    </a>
     for more details 
  </p>
</DocumentFragment>
`;
